clue.js

```diff
@@ -0,0 +1,340 @@
+]
+
+// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
+const randomSelector = array => {
+  return array[Math.floor(Math.random() * array.length)];
+}
+
+// Object to hold The Mystery
+const mystery = {
+  killer: '',
+  weapon: '',
+  room: ''
+}
+
+// Funtions to select a random Killer, Weapon, and Room to add to Mystery Object and display card.
+const pickKiller = () => {
+  mystery.killer = randomSelector(suspects)
+
+  const killer = document.getElementById('killer-card')
+  const killerName = document.getElementById("killerTitle")
+  const killerPic = document.getElementById("killerImg")
+  const killerFullName = document.getElementById("killerName")
+  const killerAge = document.getElementById("killerAge")
+  const killerWrk = document.getElementById("killerOccupation")
+  const killerFavWeapon = document.getElementById("killerFavWeapon")
+  const killerDescrip = document.getElementById("killerDescription")
+
+  killer.style.background = mystery.killer.color
+  killerName.innerHTML = mystery.killer.title
+  killerPic.src = mystery.killer.image
+  killerFullName.innerHTML = `AKA: ${mystery.killer.firstName} ${mystery.killer.lastName}`
+  killerAge.innerHTML = `Age: ${mystery.killer.age}`
+  killerWrk.innerHTML = `Occupation: ${mystery.killer.occupation}`
+  killerFavWeapon.innerHTML = `Favourite Weapon: ${mystery.killer.favoriteWeapon}`
+  killerDescrip.innerHTML = mystery.killer.description
+}
+
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+
+  const weapon = document.getElementById("weapon-card")
+  const weaponName = document.getElementById("weaponName")
+  const weaponPic = document.getElementById('weaponImg')
+  const weaponDesc = document.getElementById("weaponDescription")
+
+  weapon.style.background = mystery.weapon.color
+  weaponName.innerHTML = mystery.weapon.name
+  weaponPic.src = mystery.weapon.image
+  weaponDesc.innerHTML = mystery.weapon.description
+}
+
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms)
+
+  const room = document.getElementById("room-card")
+  const rooomName = document.getElementById("roomName")
+  const roomPic = document.getElementById('roomImg')
+
+  room.style.background = mystery.room.color
+  rooomName.innerHTML = mystery.room.name
+  roomPic.src = mystery.room.image
+}
+
+const revealMystery = () => {
+  const mysterySolution = document.getElementById("mystery")
+  const hideReveal = document.getElementById("mystery-reveal")
+  const showReset = document.getElementById("reset")
+  if (!mystery.killer || !mystery.room || !mystery.weapon){
+    mysterySolution.innerHTML = 'Please choose one card from every deck.'
+  } else {
+    mysterySolution.innerHTML = 
+    `It was ${mystery.killer.title} with the ${mystery.weapon.name} in the ${mystery.room.name}!`
+    hideReveal.style.display = "none"
+    showReset.style.display = "block"
+  }
+}  
+
+function reset() {
+  const startOver = location.reload();
+}
```

Comment:
Why do we save `startOver`?

Reply:
Hmmm! I moved `location.reload()` into the onclick attribute in the HTML. Is that a better place to have it?